src/components/ArtifactDetailedInfo.tsx

```diff
@@ -248,6 +261,259 @@ const ArtifactDetailedInfoKojiBuild: React.FC<
     );
 };
 
+interface ArtifactDetailedInfoModuleBuildProps {
+    artifact: Artifact;
+}
+
+const ArtifactDetailedInfoModuleBuild: React.FC<
+    ArtifactDetailedInfoModuleBuildProps
+> = ({ artifact }) => {
+    const [activeTabKey, setActiveTabKey] = useState<string>('build-info');
+    const handleTabClick: TabClickHandlerType = (_event, tabIndex) => {
+        setActiveTabKey(tabIndex.toString());
+    };
+    const instance = koji_instance(artifact.type);
+
+    const { loading, data } = useQuery<ArtifactsDetailedInfoModuleBuildData>(
+        ArtifactsDetailedInfoModuleBuild,
+        {
+            variables: {
+                build_id: _.toNumber(artifact.aid),
+                mbs_instance: instance,
+                koji_instance: instance,
+                distgit_instance: instance,
+            },
+            errorPolicy: 'all',
+            notifyOnNetworkStatusChange: true,
+        },
+    );
+
+    if (loading) {
+        return (
+            <Flex className="pf-u-p-lg">
+                <FlexItem>
+                    <Spinner className="pf-u-mr-md" size="md" /> Loading build
+                    info…
+                </FlexItem>
+            </Flex>
+        );
+    }
+
+    const haveData = !loading && data && !_.isEmpty(data.mbs_build);
+    if (!haveData) {
+        return (
+            <Flex className="pf-u-p-lg">
+                <Alert
+                    isInline
+                    isPlain
+                    title="No build information available"
+                    variant="info"
+                />
+            </Flex>
+        );
+    }
+
+    const build = data.mbs_build;
+    if (_.isNil(build)) {
+        console.error('No build found in data.');
+        return null;
+    }
+
+    const moduleName = build.name;
+
+    /* Time of build */
+    const buildTimeLocal = moment(build.time_completed).local();
+    const buildTimeWithTz = buildTimeLocal.format('YYYY-MM-DD HH:mm ZZ');
+    /* Time of commit */
+    let commitTimeWithTz = 'n/a';
+    if (build.commit) {
+        const commitTimeLocal = moment
+            .unix(build.commit.committer_date_seconds)
+            .local();
+        if (commitTimeLocal.isValid())
+            commitTimeWithTz = commitTimeLocal.format('YYYY-MM-DD HH:mm ZZ');
+    }
+
+    const gitCommitLink = build.scmurl && (
+        <ExternalLink
+            className={styles['buildInfoCommitHash']}
+            href={mkLinkPkgsDevelFromSource(build.scmurl, instance)}
+        >
+            {mkCommitHashFromSource(build.scmurl)}
+        </ExternalLink>
+    );
+
+    const modulemdLink = build.scmurl && (
+        <ExternalLink
+            href={mkLinkFileInGit(
+                moduleName,
+                'modules',
+                // Let's assume the Git URL is correct for now.
```

Comment:
@mgrabovsky please provide more info about this comment.
can be git-url incorrect?

Reply:
Well, the thing is that `mkCommitHashFromSource()` can return `undefined` in case it fails to parse the URL. It shouldn't happen but the type allows for the possibility.

Reply:
ah, ok.